pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitStatusDao.java

```diff
@@ -160,6 +161,9 @@ public NonPepperKitStatusDto build(ResultSet foundKitResults, Map<Integer, UserD
                                 NonPepperStatusKitService.convertTimeStringIntoTimeStamp(foundKitResults.getLong(DBConstants.DISCARD_DATE)))
                         .withDiscardBy(
                                 NonPepperStatusKitService.getUserEmailForFields(foundKitResults.getString(DBConstants.DISCARD_BY), users))
+                        .withCurrentStatus(NonPepperStatusKitService.calculateCurrentStatus(foundKitResults))
```

Comment:
This other field is calculated based on other values

Reply:
makes sense now that you point it, I will move it 
I was fooled by this method not running any direct query